wp-admin/js/widgets/media-video-widget.js

```diff
@@ -68,47 +68,44 @@
 		/**
 		 * Get the external video thumbnail for the preview
 		 *
-		 * @returns {Promise} CORS response
+		 * @returns {Promise} Promise that resolves with oEmbed object containing thumbnail_url.
 		 */
 		getExternalThumbnail: function getExternalThumbnail() {
-			var control = this, url, isYouTube, isVimeo, deffered;
-			url = control.model.get( 'url' );
-			isYouTube = url.match( /youtube|youtu\.be/ );
-			isVimeo = -1 !== url.indexOf('vimeo');
-			deffered = $.Deferred();
+			var control = this, urlParser = document.createElement( 'a' );
```

Comment:
Great refactor here, thanks!  I had grabbed much of the code above from `wp_underscore_video_template`, but really love how this method reads now, nice work 🌟